File: server/internal/domain/dto/otp.go
package dto

import (
	"time"
)

type ListOTPsResponse struct {
	ID        int    `json:"Id"`
	Issuer    string `json:"Issuer"`
	Label     string `json:"Label"`
	Algorithm string `json:"Algorithm"`
	Digits    int    `json:"Digits"`
	Period    int    `json:"Period"`
	Counter   int    `json:"Counter"`
	Method    string `json:"Method"`
}

type GenerateOTPCodesResponse struct {
	ID              int       `json:"Id"`
	CurrentCode     string    `json:"CurrentCode"`
	CurrentExpireAt time.Time `json:"CurrentExpireAt"`
	NextCode        string    `json:"NextCode"`
	NextExpireAt    time.Time `json:"NextExpireAt"`
}

File: client/types/otp.ts
export interface OTP {
  Id: string;
  Issuer: string;
  Label: string;
  Secret: string;
  Period: number;
}

export interface OTPSecret {
  Id: string;
  CurrentCode: string;
  CurrentExpireAt: Date;
  NextCode: string;
  NextExpireAt: Date;
}
